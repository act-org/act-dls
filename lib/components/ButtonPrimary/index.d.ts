/**
 * @prettier
 */
import * as React from 'react';
import { Props as ButtonBaseProps } from '~/components/ButtonBase';
export declare type Props = ButtonBaseProps;
declare const ButtonPrimary: React.FC<ButtonBaseProps>;
export default ButtonPrimary;
//# sourceMappingURL=index.d.ts.map