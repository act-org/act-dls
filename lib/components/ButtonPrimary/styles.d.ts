/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"label" | "text" | "outlined", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map