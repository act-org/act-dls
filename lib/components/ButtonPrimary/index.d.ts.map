{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/ButtonPrimary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAE/B,OAAmB,EAAE,KAAK,IAAI,eAAe,EAAE,MAAM,yBAAyB,CAAC;AAI/E,oBAAY,KAAK,GAAG,eAAe,CAAC;AAEpC,QAAA,MAAM,aAAa,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAMlC,CAAC;AAEF,eAAe,aAAa,CAAC"}