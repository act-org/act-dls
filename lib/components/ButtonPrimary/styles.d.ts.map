{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/ButtonPrimary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAIH,wBAoBI"}