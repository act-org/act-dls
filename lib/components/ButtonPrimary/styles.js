"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _makeStyles = _interopRequireDefault(require("../../styles/makeStyles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * @prettier
 */
var _default = (0, _makeStyles["default"])(function (_ref) {
  var palette = _ref.palette,
      shape = _ref.shape,
      typography = _ref.typography;
  return {
    label: {
      textTransform: 'none'
    },
    outlined: {
      border: "2px solid red",
      borderRadius: shape.borderRadius * 5,
      color: palette.primary.main,
      fontSize: typography.fontSize,
      fontWeight: typography.fontWeightMedium
    },
    text: {
      '& .MuiButton-endIcon': {
        marginLeft: 'auto'
      },
      '& .MuiButton-startIcon': {
        color: palette.primary.main
      },
      justifyContent: 'flex-start'
    }
  };
});

exports["default"] = _default;