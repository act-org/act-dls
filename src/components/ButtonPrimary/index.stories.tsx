/**
 * @prettier
 */

import * as React from 'react';

import ButtonPrimary from '.';

export const defaultAppearance = (): React.ReactElement<any> => (
  <ButtonPrimary>Primary</ButtonPrimary>
);

export default { title: 'ButtonPrimary' };
