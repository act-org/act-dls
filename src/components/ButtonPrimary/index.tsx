/**
 * @prettier
 */

import * as React from 'react';

import ButtonBase, { Props as ButtonBaseProps } from '~/components/ButtonBase';

import useStyles from './styles';

export type Props = ButtonBaseProps;

const ButtonPrimary: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => {
  const classes = useStyles();

  return <ButtonBase classes={classes} variant="outlined" {...props} />;
};

export default ButtonPrimary;
