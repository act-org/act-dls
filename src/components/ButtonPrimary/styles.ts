/**
 * @prettier
 */

import makeStyles from '~/styles/makeStyles';

export default makeStyles(({ palette, shape, typography }: any) => ({
  label: {
    textTransform: 'none',
  },
  outlined: {
    border: `2px solid red`,
    borderRadius: shape.borderRadius * 5,
    color: palette.primary.main,
    fontSize: typography.fontSize,
    fontWeight: typography.fontWeightMedium,
  },
  text: {
    '& .MuiButton-endIcon': {
      marginLeft: 'auto',
    },
    '& .MuiButton-startIcon': {
      color: palette.primary.main,
    },
    justifyContent: 'flex-start',
  },
}));
