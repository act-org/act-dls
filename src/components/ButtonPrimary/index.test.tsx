/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import ButtonPrimary from '.';

describe('ButtonPrimary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ButtonPrimary onClick={noop}>Button Label</ButtonPrimary>,
    );

    expect(container).toMatchSnapshot();
  });
});
