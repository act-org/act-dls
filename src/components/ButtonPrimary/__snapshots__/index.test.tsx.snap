// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-outlined-2"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label makeStyles-label-1"
    >
      Button Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
